sdk/identity/identity-common/review/identity-common.api.md

```diff
@@ -0,0 +1,504 @@
+export function getIdentityTokenEndpointSuffix(tenantId: string): string;
+
+// @public
+export function getKnownAuthorities(tenantId: string, authorityHost: string): string[];
+
+// @public
+export function getMSALLogLevel(logLevel: AzureLogLevel | undefined): msalCommon.LogLevel;
+
+// @public
+export class IdentityClient extends ServiceClient implements INetworkModule {
+    constructor(options?: TokenCredentialOptions);
+    // (undocumented)
+    abortRequests(correlationId?: string): void;
+    // (undocumented)
+    authorityHost: string;
+    // (undocumented)
+    generateAbortSignal(correlationId: string): AbortSignalLike;
+    // (undocumented)
+    getCorrelationId(options?: NetworkRequestOptions): string;
+    // (undocumented)
+    refreshAccessToken(tenantId: string, clientId: string, scopes: string, refreshToken: string | undefined, clientSecret: string | undefined, options?: GetTokenOptions): Promise<TokenResponse | null>;
+    // (undocumented)
+    sendGetRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<NetworkResponse<T>>;
+    // (undocumented)
+    sendPostRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<NetworkResponse<T>>;
+    // (undocumented)
+    sendTokenRequest(request: PipelineRequest): Promise<TokenResponse | null>;
+}
+
+// @public
+export type IdentityPlugin = (context: unknown) => void;
+
+// @public
+export function logEnvVars(credentialName: string, supportedEnvVars: string[]): void;
+
+// @public
+export const logger: AzureLogger;
+
+// @public
+export class MsalBaseUtilities {
+    constructor(options: MsalFlowOptions);
+    // (undocumented)
+    protected account: AuthenticationRecord | undefined;
+    generateUuid(): string;
+    protected handleError(scopes: string[], error: Error, getTokenOptions?: GetTokenOptions): Error;
+    protected handleResult(scopes: string | string[], clientId: string, result?: MsalResult, getTokenOptions?: GetTokenOptions): AccessToken;
+    // (undocumented)
+    protected logger: CredentialLogger;
+}
+
+// @public
+export interface MsalFlow {
+    getActiveAccount(): Promise<AuthenticationRecord | undefined>;
+    getToken(scopes?: string[], options?: CredentialFlowGetTokenOptions): Promise<AccessToken>;
+    getTokenSilent(scopes?: string[], options?: CredentialFlowGetTokenOptions): Promise<AccessToken>;
+    init(options?: CredentialFlowGetTokenOptions): Promise<void>;
+}
+
+// @public
+export interface MsalFlowOptions {
+    // (undocumented)
+    authenticationRecord?: AuthenticationRecord;
+    // (undocumented)
+    authorityHost?: string;
+    // (undocumented)
+    clientId?: string;
+    // (undocumented)
+    disableAutomaticAuthentication?: boolean;
+    // (undocumented)
+    getAssertion?: () => Promise<string>;
+    // (undocumented)
+    logger: CredentialLogger;
+    // (undocumented)
+    tenantId?: string;
+}
+
+// @public
+export abstract class MsalNode extends MsalBaseUtilities implements MsalFlow {
```

Comment:
I think my meta-question for this package is how did we determine what all to hoist into it? 

I know we needed to put at least the plugin stuff and related interfaces, and I think I understand that there are some useful common helpers (e.g. formatters, loggers, error shapes) that we want to be consistent across plugins, but were there other goals? 

Exposing the MSAL wrapper details will make it a little trickier to refactor those credentials in the future, for example.

Reply:
That was my main intention as well. I didn't want to pull the MSAL wrapper into the common package. But the tests in the plugin packages were referencing some of these MSAL wrappers for types for creating tests and this is why I had to pull in these wrappers into identity-common. In case the tests could be written without referencing the wrappers then we won't have to do this. cc: @willmtemple

Reply:
I think we need a different type of abstraction for the testing components. For those it would be sufficient to have a local devDep that we don't publish rather than putting everything into a public package. `identity-test` or some such.